### PR TITLE
Added historyindexcachesize to class zabbix

### PR DESCRIPTION
**Pull request by Ordnaxz**
Fixes https://github.com/voxpupuli/puppet-zabbix/issues/566

As explained, historyindexcachesize was missing from the zabbix class, only making it usable under zabbix::server.

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @Ordnaxz, thanks for the patch! Can you take a look at the used email address in the commit? It isn't associated with your github account. Can you also add a unit test for this? We've got many examples in https://github.com/voxpupuli/puppet-zabbix/tree/master/spec/classes that you can update.

File: manifests/params.pp [hunk 1]
```diff
@@ -161,7 +161,7 @@
   $server_debuglevel                        = '3'
   $server_externalscripts                   = '/usr/lib/zabbix/externalscripts'
   $server_historycachesize                  = '8M'
-  $server_historyindexcachesize             = undef
+  $server_historyindexcachesize             = '4M'
```
Comment (bastelfreak) [inline manifests/params.pp]: can the default stay at undef? Otherwise it would be a breaking change that requires a major release (instead of a minor).

Comment (Ordnaxz) [reply manifests/params.pp]: Sure can! I'll revert it.

**Comment by Ordnaxz** [comment]
@bastelfreak I've reverted the changes and moved the tests to Zabbix 3.0 and 3.2 (as pre-3.0 did not have historyindexcachesize).

File: manifests/init.pp [hunk 2]
```diff
@@ -145,6 +145,7 @@
   $cacheupdatefrequency                     = $zabbix::params::server_cacheupdatefrequency,
   $startdbsyncers                           = $zabbix::params::server_startdbsyncers,
   $historycachesize                         = $zabbix::params::server_historycachesize,
+  $historyindexcachesize                    = $zabbix::params::server_historyindexcachesize,
```
Comment (bastelfreak) [inline manifests/init.pp]: can you please add a dataype to the parameter?

**Comment by bastelfreak** [comment]
Hi @Ordnaxz, thanks for the updates! Just a tiny thing left, could you please add a dataype for the new parameter and rebase against master? Afterwards we can merge this.

File: manifests/init.pp (hunk 2 above)
Comment (Ordnaxz) [reply manifests/init.pp]: Done.

**Comment by Ordnaxz** [comment]
Hello @bastelfreak, I've added the data type and rebased to the current master branch. Travis CI test seems to be failing now though, not sure why.. It's also failing in the current master branch though, so I suppose it's not on my side.

**Comment by bastelfreak** [comment]
Hi @Ordnaxz, your branch wasn't properly rebased against our master. If you need some help with git feel free to join our IRC channel #voxpupuli on freenode or our slack channel at https://slack.puppet.com

**Comment by Ordnaxz** [comment]
@bastelfreak Rather than attempting to undo my mess and fix it, I've opened a new pull request with a new branch. Please see: https://github.com/voxpupuli/puppet-zabbix/pull/586

I've mirrored all the changes I've made in this pull request.

**Pull request closed without merging**